homeassistant/components/tami4/sensor.py

```diff
@@ -0,0 +1,127 @@
+from .entity import Tami4EdgeBaseEntity
+
+_LOGGER = logging.getLogger(__name__)
+
+ENTITY_DESCRIPTIONS = [
+    SensorEntityDescription(
+        key="uv_last_replacement",
+        translation_key="uv_last_replacement",
+        icon="mdi:calendar",
+        device_class=SensorDeviceClass.DATE,
+    ),
+    SensorEntityDescription(
+        key="uv_upcoming_replacement",
+        translation_key="uv_upcoming_replacement",
+        icon="mdi:calendar",
+        device_class=SensorDeviceClass.DATE,
+    ),
+    SensorEntityDescription(
+        key="uv_status",
+        translation_key="uv_status",
+        icon="mdi:clipboard-check-multiple",
+    ),
+    SensorEntityDescription(
+        key="filter_last_replacement",
+        translation_key="filter_last_replacement",
+        icon="mdi:calendar",
+        device_class=SensorDeviceClass.DATE,
+    ),
+    SensorEntityDescription(
+        key="filter_upcoming_replacement",
+        translation_key="filter_upcoming_replacement",
+        icon="mdi:calendar",
+        device_class=SensorDeviceClass.DATE,
+    ),
+    SensorEntityDescription(
+        key="filter_status",
+        translation_key="filter_status",
+        icon="mdi:clipboard-check-multiple",
+    ),
+    SensorEntityDescription(
+        key="filter_litters_passed",
+        translation_key="filter_litters_passed",
+        icon="mdi:water",
+        state_class=SensorStateClass.TOTAL,
+        device_class=SensorDeviceClass.WATER,
+        native_unit_of_measurement=UnitOfVolume.LITERS,
+    ),
+]
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
+) -> None:
+    """Perform the setup for Tami4Edge."""
+    data = hass.data[DOMAIN][entry.entry_id]
+    api: Tami4EdgeAPI = data[DATA_API]
+    coordinator: Tami4EdgeWaterQualityCoordinator = data[DATA_COORDINATOR]
+
+    entities = []
+    for entity_description in ENTITY_DESCRIPTIONS:
+        entities.append(
+            Tami4EdgeSensorEntity(
+                coordinator=coordinator,
+                api=api,
+                entity_description=entity_description,
+            )
+        )
+
+    async_add_entities(entities)
+
+
+class Tami4EdgeSensorEntity(
+    Tami4EdgeBaseEntity,
+    CoordinatorEntity[Tami4EdgeWaterQualityCoordinator],
+    SensorEntity,
+):
+    """Representation of the entity."""
+
+    def __init__(
+        self,
+        coordinator: Tami4EdgeWaterQualityCoordinator,
+        api: Tami4EdgeAPI,
+        entity_description: SensorEntityDescription,
+    ) -> None:
+        """Initialize the Tami4Edge sensor entity."""
+        Tami4EdgeBaseEntity.__init__(self, api, entity_description)
+        CoordinatorEntity.__init__(self, coordinator)
+
+    @property
+    def native_value(self) -> date | str | float:
+        """Return the state of the device."""
+        return getattr(self.coordinator.data, self.entity_description.key)  # type: ignore[no-any-return]
```

Comment:
```suggestion
```

Can be removed as the value is already set in `def _handle_coordinator_update(self) -> None:`

Reply:
For some reason the state shows as `Unknown` when not having the `native_value` property set

Reply:
See comment about above

Reply:
When changing `async_add_entities(entities)` into `async_add_entities(entities, True)`, the coordinator `_async_update_data` is being called twice.
First time by [`await coordinator.async_config_entry_first_refresh()` in `__init__.py`](https://github.com/home-assistant/core/pull/90056/files#diff-d67f53e1a9e63fde874e2f1a6649251134682844db6df877138a644c42b8c7c2R29) and second time by the `async_add_entities`.

I can either remove the `async_config_entry_first_refresh` and use `async_add_entities(entities, True)`,
or keep `async_config_entry_first_refresh` and use `native_value`.